integration-tests/reorg/reorg_confirmer.go

```diff
@@ -117,7 +117,7 @@ func (rc *ReorgController) WaitReorgStarted() {
 func (rc *ReorgController) ReceiveBlock(blk blockchain.NodeBlock) error {
 	rc.mutex.Lock()
 	defer rc.mutex.Unlock()
-	if blk.Block == nil {
+	if blk.Block.Number() == nil {
```

Comment:
You might be able to remove these checks altogether now that blocks are passed by value. Unless there's some StarkNet edge case where this can happen?

Reply:
great! i'll remove it - i only rewrote the check in the case that it was still needed, but it was meant to function as a nil pointer check, then yeah it can be removed